server/node/src/main/java/org/kaaproject/kaa/server/control/service/DefaultControlService.java

```diff
@@ -1015,6 +1019,59 @@ public void onNodeAdded(OperationsNodeInfo node) {
     return resolver.getNode(entityId);
   }
 
+  /**
+   * Use when one need to write an OperationsNodeInfo instance to logs. The OperationsNodeInfo
+   * instance has ByteBuffer fields which should be represented in log files as null because their
+   * values are unrepresentative and redundant.
+   *
+   * @param format   the format string
+   * @param node     the instance of OperationsNodeInfo
+   * @param resolver the instance of OperationsServerResolver
+   */
+  private void writeLogWithoutByteBuffer(String format,
+                                         OperationsNodeInfo node,
```

Comment:
I think this method should be rewritten using Java 8. Now the code looks so hard